File: Electronics_Shop_17.Services/Database/Warranty.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Services.Database
{
    public class Warranty : IHelperId
    {
        public int Id { get; set; }
        public int Period_mm { get; set; }
        public string CoverageDetails { get; set; }
        public int ProductId { get; set; }
        //public virtual Product Product { get; set; }

    }
}

File: angular17_web/src/app/models/warranty/warranty.ts
export class Warranty {
  id!: number;
  period_mm!: number;
  coverageDetails!: string;
  productId!: number;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.period_mm = data.period_mm;
    this.coverageDetails = data.coverageDetails;
    this.productId = data.productId;
  }
}
